crypto/kdf/hkdf.c

```diff
@@ -71,18 +71,19 @@ static int pkey_hkdf_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
     HKDF_PKEY_CTX *kctx = ctx->data;
 
     switch (type) {
-    case EVP_PKEY_CTRL_HKDF_MD:
+    case EVP_PKEY_CTRL_KDF_MD:
+    case EVP_PKEY_CTRL_OLD_HKDF_MD:
         if (p2 == NULL)
             return 0;
 
         kctx->md = p2;
         return 1;
 
-    case EVP_PKEY_CTRL_HKDF_MODE:
+    case EVP_PKEY_CTRL_KDF_MODE:
```

Comment:
Some other KDF might have the concept of a "mode" but the meaning is likely to be completely different and it might not take an int parameter. For HKDF there are the three permitted values EVP_PKEY_HKDEF_MODE_EXTRACT_AND_EXPAND, EVP_PKEY_HKDEF_MODE_EXTRACT_ONLY and EVP_PKEY_HKDEF_MODE_EXPAND_ONLY that is likely to be completely different if the KDF is not HKDF.